File: 승준/20230714/decorating_the_rooftop_garden_6198.js
const fs = require("fs");

const input=fs.readFileSync('/dev/stdin').toString().trim().split('\n').map((line) => line.trim()).map(Number);
// const input = fs
//   .readFileSync("input.txt")
//   .toString()
//   .trim()
//   .split("\n")
//   .map((line) => line.trim())
//   .map(Number);

const [n, ...buildings] = input;

let count = 0
let stack = []
for (b of buildings) {
    while (stack.length > 0 && stack[stack.length-1] <= b ) {
        stack.pop()
    }
    stack.push(b)
    count += stack.length - 1
}

console.log(count)

File: 승준/20230714/decorating_the_rooftop_garden_6198.py
import sys
input = sys.stdin.readline

N = int(input())
buildings = [int(input()) for _ in range(N)]

stack = []
count = 0

for b in buildings:
    while stack and stack[-1] <= b:
        stack.pop()
    stack.append(b)
    
    count += len(stack) - 1
    
print(count)